kroxylicious/src/main/java/io/kroxylicious/proxy/bootstrap/FilterChainFactory.java

```diff
@@ -25,8 +27,23 @@ public class FilterChainFactory {
     private final Configuration config;
 
     public FilterChainFactory(Configuration config) {
-        Objects.requireNonNull(config);
-        this.config = config;
+        this.config = Objects.requireNonNull(config);
+    }
+
+    public static boolean validateFilterConfiguration(List<FilterDefinition> filters) {
```

Comment:
Yeah :/ 

It feel very wrong having a unit test that relied on the method completing without an exception so I gave it a return value but as you say thats still not very satisfying. 

The problem with returning `false` rather than throwing is you loose all context as to which filters failed the check. Maybe it should return the set of failing filters and let the callers throw/test for empty/non empty?

Reply:
maybe it could be reworked to be 

`public static boolean isValid(FilterDefinition filter) {`

And leave it up to the caller to do `Collectors.partitioningBy(FilterChainFactor::isValid)` plus the failure logic?

Reply:
though you lose the validity reason

Reply:
could introduce a Result class for that (why doesn't the JVM have one!)

Reply:
Yeah, I think the invalid reason is quite important.

My current preference is to move to something like 
`public static Set<FilterDefinition> validateFilterConfiguration(List<FilterDefinition> filters)` where the return set is the invalid filters.

Reply:
resolved with 9ed2be715736f8eef51502a00fa8b123739b693f